src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy

```diff
@@ -1280,9 +1343,11 @@ public enum DarknessAblaze implements LogicCardInfo {
         move "Deep Sea Swirl", {
           text "Shuffle your hand into your deck, then draw 8 cards."
           energyCost C
+          // TODO: Does this need deck to not be empty?
```

Comment:
Hmm... On current implementation of cards, I'd say you could add an assert for `my.deck || my.hand` (Not 100% sure it'd be required tho). But you can definitely use the attack if you have hand, since you'd shuffle that into the deck and then draw the 8 (or the whole hand again).

Reply:
Right. I figured Deck counts as non-public information, so shuffling into it, even if the end result is drawing all of the cards probably still counts as an effect happening. I will add that assertion though since in the case both are 0 there definitely isn't an actual effect.